### PR TITLE
[SYCL] Added error handling for non-uniform work group size case

### PR DESCRIPTION
**Pull request by etdamiba-intel**
Signed-off-by: Eugene T Damiba <eugene.t.damiba@intel.com>

### CODE REVIEW

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -2953,11 +2953,20 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
...
+  if(GlobalWorkSize[0] != (ZeThreadGroupDimensions.groupCountX * WG[0])) {
+    zePrint("piEnqueueKernelLaunch: invalid work_dim\n");
+    return PI_INVALID_WORK_GROUP_SIZE;
+  }
+  if(GlobalWorkSize[1] != (ZeThreadGroupDimensions.groupCountY * WG[1])) {
+    zePrint("piEnqueueKernelLaunch: invalid work_dim\n");
+    return PI_INVALID_WORK_GROUP_SIZE;
+  }
+  if(GlobalWorkSize[2] != (ZeThreadGroupDimensions.groupCountZ * WG[2])) {
+    zePrint("piEnqueueKernelLaunch: invalid work_dim\n");
```
Comment (keryell) [inline sycl/plugins/level_zero/pi_level_zero.cpp:2966]: Please add some helpful message for the end-user, like:
```suggestion
    zePrint("piEnqueueKernelLaunch: invalid work_dim: the range is not a multiple of the group size in the 3rd dimension\n");
```
Add the same for the  other dimensions.

**Review by keryell — CHANGES_REQUESTED** [review]
Please help the programmers! :-)
This is a low-hanging fruit...

File: sycl/plugins/level_zero/pi_level_zero.cpp (hunk 1 above)
Comment (etdamiba-intel) [reply sycl/plugins/level_zero/pi_level_zero.cpp:2966]: @keryell ,Thank you for the feedback. I have incorporated the proposed changes.

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 2]
```diff
@@ -2956,15 +2956,18 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
 
   // Error handling for non-uniform group size case
   if (GlobalWorkSize[0] != (ZeThreadGroupDimensions.groupCountX * WG[0])) {
-    zePrint("piEnqueueKernelLaunch: invalid work_dim. The range is not a multiple of the group size in the 3rd dimension\n");
+    zePrint("piEnqueueKernelLaunch: invalid work_dim. The range is not a "
+            "multiple of the group size in the 3rd dimension\n");
```
Comment (keryell) [inline sycl/plugins/level_zero/pi_level_zero.cpp:2960]:
```suggestion
            "multiple of the group size in the 1st dimension\n");
```

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 3]
```diff
@@ -2956,15 +2956,18 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
...
     return PI_INVALID_WORK_GROUP_SIZE;
   }
   if (GlobalWorkSize[1] != (ZeThreadGroupDimensions.groupCountY * WG[1])) {
-    zePrint("piEnqueueKernelLaunch: invalid work_dim. The range is not a multiple of the group size in the 3rd dimension\n");
+    zePrint("piEnqueueKernelLaunch: invalid work_dim. The range is not a "
+            "multiple of the group size in the 3rd dimension\n");
```
Comment (keryell) [inline sycl/plugins/level_zero/pi_level_zero.cpp:2965]:
```suggestion
            "multiple of the group size in the 2nd dimension\n");
```

**Review by keryell — CHANGES_REQUESTED** [review]
Almost programmer friendly! :-)

**Pull request merged**